### PR TITLE
Added multiple related fixes to enable automatic addition of KeyToValue

### PR DESCRIPTION
**Pull request by harishsk**
This PR includes a number of fixes to enable automatic addition of KeyToValue in the Nimbus codepath. Specifically:
* Fixed BinaryClassfierScorer to support exporting key types
* Fixed PredictedLabelScorerBase to include the right types and shapes in the onnx graph
* Fixed OneVersusAllTrainer to include shape information
* Fixed SaveAsOnnxCommand to include type information in the last Identity node
* Fixed OptionalColumnTransform to have the correct variable names and types. (This one is a preemptive fix. It fixes a crash that happens with the code in the ORT master branch)

### CODE REVIEW

File: src/Microsoft.ML.Data/Scorers/BinaryClassifierScorer.cs [hunk 1]
```diff
@@ -199,23 +199,26 @@ private protected override void SaveAsOnnxCore(OnnxContext ctx)
             for (int iinfo = 0; iinfo < Bindings.InfoCount; ++iinfo)
                 outColumnNames[iinfo] = Bindings.GetColumnName(Bindings.MapIinfoToCol(iinfo));
 
-            string opType = "Binarizer";
+            string scoreColumn = Bindings.RowMapper.OutputSchema[Bindings.ScoreColumnIndex].Name;
+
             OnnxNode node;
-            var binarizerOutput = ctx.AddIntermediateVariable(null, "BinarizerOutput", true);
+            string opType = "Binarizer";
+            var binarizerOutput = ctx.AddIntermediateVariable(NumberDataViewType.Single, "BinarizerOutput", false);
```
Comment (ganik) [inline src/Microsoft.ML.Data/Scorers/BinaryClassifierScorer.cs]: >false [](start = 108, length = 5)

What does this false mean?

Comment (harishsk) [reply src/Microsoft.ML.Data/Scorers/BinaryClassifierScorer.cs]: False = Do not skip adding shape and type information (or in other words, add shape and type information). 

---
In reply to: [383088227](https://github.com/dotnet/machinelearning/pull/4878#discussion_r383088227) [](ancestors = 383088227)

File: src/Microsoft.ML.Data/Scorers/BinaryClassifierScorer.cs [hunk 2]
```diff
@@ -199,23 +199,26 @@ private protected override void SaveAsOnnxCore(OnnxContext ctx)
...
-            string scoreColumn;
-            if (Bindings.RowMapper.OutputSchema[Bindings.ScoreColumnIndex].Name == "Score")
-                scoreColumn = outColumnNames[1];
-            else
+            string comparisonOutput = binarizerOutput;
+            if (Bindings.PredColType is KeyDataViewType)
             {
-                Host.Assert(Bindings.InfoCount >= 3);
-                scoreColumn = outColumnNames[2];
+                var one = ctx.AddInitializer(1.0f, "one");
```
Comment (ganik) [inline src/Microsoft.ML.Data/Scorers/BinaryClassifierScorer.cs]: >ctx.AddInitializer(1.0f, "one"); [](start = 26, length = 32)

Is this +1 ? to be 1 based?

Comment (harishsk) [reply src/Microsoft.ML.Data/Scorers/BinaryClassifierScorer.cs]: Yes, this is to make it one based and make it consistent with ML.NET results.

---
In reply to: [383088629](https://github.com/dotnet/machinelearning/pull/4878#discussion_r383088629) [](ancestors = 383088629)

File: src/Microsoft.ML.Data/Transforms/ValueToKeyMappingTransformer.cs [hunk 3]
```diff
@@ -804,7 +804,7 @@ private bool SaveAsOnnxCore(OnnxContext ctx, int iinfo, ColInfo info, string src
                 else if (info.TypeSrc.GetItemType().Equals(NumberDataViewType.Double))
                 {
                     // LabelEncoder doesn't support double tensors, so values are cast to floats
-                    var castOutput = ctx.AddIntermediateVariable(null, "castOutput", true);
```
Comment (ganik) [inline src/Microsoft.ML.Data/Transforms/ValueToKeyMappingTransformer.cs]: >null [](start = 65, length = 4)

How did it work with null before? Was there an exception?

Comment (harishsk) [reply src/Microsoft.ML.Data/Transforms/ValueToKeyMappingTransformer.cs]: If the last parameter is true, it skips adding shape and type information and therefore accepts null. This still works, but I am prepping some parts of the code base for issues I have seen when run against the master branch of ORT. 

---
In reply to: [383088824](https://github.com/dotnet/machinelearning/pull/4878#discussion_r383088824) [](ancestors = 383088824)

**Pull request merged**